test/mysql-cdc/table-in-mysql-schema.td

```diff
@@ -0,0 +1,62 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ set-sql-timeout duration=1s
+$ set-max-tries max-tries=20
+
+$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+ALTER SYSTEM SET enable_mysql_source = true
+
+#
+# Test that tables in the mysql schema are not replicated
+#
+
+> CREATE SECRET mysqlpass AS '${arg.mysql-root-password}'
+> CREATE CONNECTION mysql_conn TO MYSQL (
+    HOST mysql,
+    USER root,
+    PASSWORD SECRET mysqlpass
+  )
+
+$ mysql-connect name=mysql url=mysql://root@mysql password=${arg.mysql-root-password}
+
+$ mysql-execute name=mysql
+USE mysql;
+
+# Insert data pre-snapshot
+CREATE TABLE t_in_mysql (f1 INT);
+INSERT INTO t_in_mysql VALUES (1), (2);
+
+! CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_conn
+  FOR ALL TABLES;
+contains:No tables found
+
+! CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_conn
+  FOR TABLES (mysql.timezone);
+contains:No tables found
+
+> CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_conn
+  FOR TABLES (mysql.t_in_mysql);
```

Comment:
Surprised that this works...

Reply:
this is intended -- we will ignore the `mysql` schema in the case of `FOR ALL TABLES` since it has a bunch of system tables and it is not possible for us to distinguish user / system tables in it. However if the user needs to ingest a table they added to that schema they can use the `FOR TABLES` syntax, which is what you are doing here